source-mongodb/pull.go

```diff
@@ -137,7 +157,7 @@ type capture struct {
 }
 
 type deprecatedCaptureState struct {
-	Resources map[string]bson.Raw `json:"resources"`
+	Resources map[string]json.RawMessage `json:"resources"`
```

Comment:
@williamhbaker `json.RawMessage` and `bson.Raw` are both `[]byte` under the hood. In case of the initial `Unmarshal`, `json.RawMessage` basically does not touch the underlying bytes and you get the original raw byte array that was found under the key.

Afterwards we just cast that `json.RawMessage` to `bson.Raw`, but the underlying data stays the same byte array

Reply:
> Afterwards we just cast that json.RawMessage to bson.Raw, but the underlying data stays the same byte array

@mdibaiee I'm a little late to the party here, but this does not sound safe at all. `json.RawMessage` is expected to be JSON-encoded data, whereas `bson.Raw` is expected to be BSON-encoded. The two are not compatible, so I don't see how this would work without parsing the JSON and converting it to BSON. I'd think a type cast would just result in an invalid `bson.Raw` value.

Reply:
Oh, that's a good catch. It seems likely that `bson.Raw` being an alias for `[]byte` is serialized into JSON as a base64-encoded string, and the `json.RawMessage` would still be that string (including double-quotes around it), so there still needs to be a step that strips those quotes and decodes base64 into bytes before casting to BSON.